ivre/db/mongo.py

```diff
@@ -1127,6 +1127,20 @@ def searchports(self, ports, protocol='tcp', state='open', neg=False):
             for port in ports]}}
 
     @staticmethod
+    def searchobjectid(oid, neg=False):
+        """Filters records by their ObjectID.
+        oid can either be a single ObjectID in string or bson.objectid.ObjectID
+        format, or an iterable of ObjectIDs"""
+        if not hasattr(oid, '__iter__'):
```

Comment:
I think a test like `isinstance(oid, basestring) or isinstance(oid, bson.objectid.ObjectId)` is more appropriate.

Also, you could handle, as you did in the previous version, iterables with only one item.

For that, you could make sure oid is a list before doing the work. For example:
```
if isinstance(oid, basestring) or isinstance(oid, bson.objectid.ObjectId):
    oid = [bson.objectid.ObjectId(oid)]
else:
    oid = [bson.objectid.ObjectId(elt) for elt in oid]
```

After that, `oid` is always a list of `ObjectId`s.